### ISSUE TITLE
Overlapping Classes Method Issue with Multilabel Data

### ISSUE THREAD

**Issue by dafajon** [cleanlab/cleanlab#653]
I have a multilabel text classification problem with 33 labels. I am using the `find_overlapping_classes` accordingly with its [documentation](https://docs.cleanlab.ai/stable/cleanlab/dataset.html#cleanlab.dataset.find_overlapping_classes)

```python
cleanlab.dataset.find_overlapping_classes(labels=true_labels, pred_probs=pred, multi_label=True)
```

`true_labels` is an iterable on size N lists that contain list of labels.
`pred_probs` is an 2d array of size N samples x C classes of probabilities. Out of fold predictions.

```python
len(true_labels), preds.shape
```
```python
(71658, (71658, 33))
```

```python
true_labels[:10]
```
`[[0], [3], [7], [11], [0, 3], [0], [3], [9], [2], [3]]`

# Stack trace

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/Users/dorukhanafacan/Documents/multilang_etl/cleanlab_label_validation.ipynb Cell 40' in <cell line: 1>()
----> [1](vscode-notebook-cell:/Users/dorukhanafacan/Documents/multilang_etl/cleanlab_label_validation.ipynb#ch0000059?line=0) cleanlab.dataset.find_overlapping_classes(labels=true_labels, pred_probs=pred, multi_label=True)

File /opt/anaconda3/envs/dsenv/lib/python3.9/site-packages/cleanlab/dataset.py:258, in find_overlapping_classes(labels, pred_probs, asymmetric, class_names, num_examples, joint, confident_joint, multi_label)
    [255](file:///opt/anaconda3/envs/dsenv/lib/python3.9/site-packages/cleanlab/dataset.py?line=254)     rcv_list = [tup for tup in rcv_list if tup[0] != tup[1]]
    [256](file:///opt/anaconda3/envs/dsenv/lib/python3.9/site-packages/cleanlab/dataset.py?line=255) else:  # symmetric
    [257](file:///opt/anaconda3/envs/dsenv/lib/python3.9/site-packages/cleanlab/dataset.py?line=256)     # Sum the upper and lower triangles and remove the lower triangle and the diagonal
--> [258](file:///opt/anaconda3/envs/dsenv/lib/python3.9/site-packages/cleanlab/dataset.py?line=257)     sym_joint = np.triu(joint) + np.tril(joint).T
    [259](file:///opt/anaconda3/envs/dsenv/lib/python3.9/site-packages/cleanlab/dataset.py?line=258)     rcv_list = _2d_matrix_to_row_column_value_list(sym_joint)
    [260](file:///opt/anaconda3/envs/dsenv/lib/python3.9/site-packages/cleanlab/dataset.py?line=259)     # Provide values only in (the upper triangle) of the matrix.

ValueError: operands could not be broadcast together with shapes (33,2,2) (2,2,33)
```
# Additional information

- **Cleanlab version**: 2.2.1
- **Operating system**: MacOS 12.5
- **Python version**:  3.9.12

**Comment by jwmueller** [comment]
Thank you for reporting this, it indeed seems like a bug in our code and we will push a fix out for it shortly!

**Comment by jwmueller** [comment]
Hi @dafajon the bug ended up being a bit more complex than anticipated, but our team just pushed a fix!
To test out the new code until the next stable release, you'll need to install the package from source via: 
```
pip install git+https://github.com/cleanlab/cleanlab.git
```

Instead of calling: `cleanlab.dataset.find_overlapping_classes(..., multi_label=True)`, in the new code you should instead:
`cleanlab.multilabel_classification.dataset.common_multilabel_issues(...)`

Please let us know if it works for you and if you have other questions!

**Comment by jwmueller** [comment]
Just FYI here are some other changes you may need to upgrade existing code to the newest version of the library (our upcoming release notes will more precisely list them all):

1)  `cleanlab.dataset.rank_classes_by_label_quality(..., multi_label=True)`
→
`cleanlab.multilabel_classification.dataset.rank_classes_by_label_quality(...)`

2) `cleanlab.dataset.find_overlapping_classes(..., multi_label=True)` 
→
`cleanlab.multilabel_classification.dataset.common_multilabel_issues(...)`

3) `cleanlab.dataset.overall_label_health_score(...multi_label=True)`
→
`cleanlab.multilabel_classification.dataset.overall_label_health_score(...)`

4) `cleanlab.dataset.health_summary(...multi_label=True)`
→ 
`cleanlab.multilabel_classification.dataset.multilabel_health_summary(...)`

These changes were made to better separate the methods specific to multi-label classification from the methods specific to multi-class classification datasets.

**Comment by jwmueller** [comment]
This has been fixed since cleanlab v2.4.0. Feel free to re-open the issue if you still encounter problems
